### PR TITLE
[EPT-1709] add top panel for list view

### PR DESCRIPTION
**Pull request by shanthru**
Adds Top Panel to display the page name and create new page button

### CODE REVIEW

**Comment by shanthru** [comment]
If we are sticking with STUI v2, should we continue using ListLayout for the time being?
See example here:
https://github.com/sailthru/hosted-pages-frontend-dev1/pull/1/commits/30ff0e671030f76dfdebd6db133bda6c04f5d801

File: src/ListView/ListViewTopPanel.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import styled from 'styled-components';
+import { PageTopPanel } from '@sailthru/stui-regions';
+import { Button } from "@sailthru/stui-elements";
+
+const Header = styled.header`
+    grid-area: header;
```
Comment (jdrury-sailthru) [inline src/ListView/ListViewTopPanel.jsx:7]: does this do anything?

i think setting position to relative would be enough

no need for a z index either unless that is something stui regions messes with

File: src/ListView/ListViewTopPanel.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    position: initial;
+    z-index: auto;
+    margin-bottom: 10px;
+`;
+
+const NewButton = styled.div`
+    display: flex;
```
Comment (jdrury-sailthru) [inline src/ListView/ListViewTopPanel.jsx:14]: does this work without flex? is this necessary

File: src/ListView/ListViewTopPanel.jsx (hunk 1 above)
Comment (shanthru) [reply src/ListView/ListViewTopPanel.jsx]: removed any unnecessary styling

File: src/ListView/ListViewTopPanel.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import styled from 'styled-components';
+import { PageTopPanel } from '@sailthru/stui-regions';
+import { Button } from "@sailthru/stui-elements";
+
+const Header = styled.header`
+    margin-bottom: 10px;
+    z-index: auto;
+`;
+
+const NewButton = styled.div`
```
Comment (jdrury-sailthru) [inline src/ListView/ListViewTopPanel.jsx:11]: no need for the extra layer here

```suggestion
const StyledButton = styled(Button)`
```

File: src/ListView/ListViewTopPanel.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    <Header>
+        <PageTopPanel
+            title="Hosted Pages"
+            backgroundColor="dark"
+        >
+            <NewButton>
+                <Button onClick={handleCreate}>
+                    New Page
+                </Button>
+            </NewButton>
```
Comment (jdrury-sailthru) [inline src/ListView/ListViewTopPanel.jsx:30]:
```suggestion
            <StyledButton onClick={handleCreate}>
                New Page
            </StyledButton>
```

File: src/App.jsx [hunk 5]
```diff
@@ -1,10 +1,12 @@
 import React from "react";
 import { ListView } from './ListView/ListView';
+import { ListViewTopPanel } from "./ListView/ListViewTopPanel";
 
 export default function App() {
   return (
-    <>
-      <ListView />
-    </>
+      <div>
```
Comment (jdrury-sailthru) [inline src/App.jsx:7]: it is fine to have the container div here but if you wanted to keep the DOM tree cleaner you could add back the fragments

the `<>`

**Review by jdrury-sailthru — APPROVED** [review]

File: src/App.jsx (hunk 5 above)
Comment (shanthru) [reply src/App.jsx]: Updated (I actually don't remember why I would have changed this)

File: src/ListView/ListViewTopPanel.jsx (hunk 3 above)
Comment (shanthru) [reply src/ListView/ListViewTopPanel.jsx]: Updated...thanks!

File: src/ListView/ListViewTopPanel.jsx (hunk 4 above)
Comment (shanthru) [reply src/ListView/ListViewTopPanel.jsx]: updated

**Pull request merged**